.github/workflows/osbuilder-checks.yaml

```diff
@@ -0,0 +1,62 @@
+# This configure jobs to build osbuilder in various ways to ensure changes
+# doesn't break the build scripts.
+#
+name: osbuilder checks
+
+on:
+  pull_request:
+    types: [opened, edited, reopened]
+    # Only run when osbuilder files changed.
+    paths:
+      - 'tools/osbuilder/**'
+
+jobs:
+  # Jobs for dracut based build.
+  dracut-build-on-fedora:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:34
+      options: --privileged
+    strategy:
+      matrix:
+        builder:
+          # Note: 'image' builder doesn't work on a container environment like
+          # the one provided by Github, so it was not added to the matrix.
+          - initrd
```

Comment:
This workflow is splitted in two job templates:

* dracut-build-on-fedora: `BUILD_METHOD` is `dracut`. I chose to use Fedora because, afaik, the only users of that approach are Fedora and Red Hat. In order to build a fedora based dracut rootfs the job must run on Fedora; GH doesn't provide a `runs-on` for Fedora, therefore, I had to instruct GH to use a `fedora:34` container on top of `runs-on: ubuntu-latest`.  On that kind of setup you cannot run nested containers (at least I couldn't make it), thus I could only build an `initrd` artifact. Nevertheless, Fedora and Red Hat build `initrd` artifacts, so here we are testing the exactly configuration required by the users of the dracut build method.   

* distro-build-with-docker: `BUILD_METHOD` is `distro`. Here the job runs on top of `runs-on: ubuntu-latest` so it can run docker, therefore, it is testing the creation of both `initrd` and `image` artifacts. However I am not running the full combination of "distro" vs "artifact type" because I believe it is enough to test all the code paths.

Note that here we aren't testing whether the resulting image/initrd works or not, but rather whether the artifact can be built or not by the different methods we support. So an alternative approach is to not have this workflow at all; instead we ensure the various build methods, artifacts types and rootfs are built and tested by jobs on Jenkins. 

That said, is there anything that I am missing here? Let me know if you think I should arrange this workflow differently.

/cc @fidencio